apps/mosaico/src/Main.purs

```diff
@@ -247,19 +262,19 @@ getDraftArticle env { params: {aptomaId}, query } = do
 
 getArticle
   :: Env
-  -> { params :: { uuidOrSlug :: String }, guards :: { credentials :: Maybe UserAuth } }
+  -> { params :: { uuidOrSlug :: String }, guards :: { credentials :: Maybe UserAuth, clientip :: Maybe String } }
   -> Aff (Response ResponseBody)
-getArticle env r@{ params: { uuidOrSlug }, guards: { credentials } }
+getArticle env r@{ params: { uuidOrSlug }, guards: { credentials, clientip } }
   | Just uuid <- UUID.parseUUID uuidOrSlug = do
       { user, article, mostReadArticles, latestArticles } <- sequential $
         { user: _, article: _, mostReadArticles: _, latestArticles: _ }
         <$> maybe (pure Nothing) (parallel <<< getUser) credentials
-        <*> parallel (Lettera.getArticle uuid mosaicoPaper r.guards.credentials)
+        <*> parallel (Lettera.getArticle uuid mosaicoPaper r.guards.credentials clientip)
```

Comment:
Let's just use `credentials clientip`. Both variables are in scope and it'd be odd style to to access one via `r` and the other directly.

Reply:
I was wondering about that